lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -721,6 +721,33 @@ def endpoints(self) -> List[Dict[str, str]]:
         return endpoints
 
 
+class PrometheusRemoteWriteAlertsChangedEvent(EventBase):
+    """Event emitted when Prometheus remote_write alerts change."""
+
+    def __init__(self, handle, relation_id):
+        super().__init__(handle)
+        self.relation_id = relation_id
+
+    def snapshot(self):
+        """Save Prometheus remote_write information."""
+        return {"relation_id": self.relation_id}
+
+    def restore(self, snapshot):
+        """Restore Prometheus remote_write information."""
+        self.relation_id = snapshot["relation_id"]
+
+
+class PrometheusRemoteWriteProviderConsumersChangedEvent(EventBase):
```

Comment:
I'm not sure this is needed for lib users.
Internally we can still do `update_endpoint` but emit `rules_changed`, because for a lib user the implication is probably the same?
I.e. less custom events to observe the better.

Reply:
This is needed for the same reason as above. Prometheus (via `_configure`) must know to actually check whether it has any relations to enable remote write or not.

Without this, Prometheus would not know (and would not enable it, or disable it) until some other event kicked `_configure`